### DISCUSSION TITLE
HOROVOD Installation Error

### DISCUSSION THREAD

**Discussion by AnuragKr** [horovod/horovod#3800] [Q&A] [answered]
Hi,

NCCL version -- 2.16.2-1
System Configuration -- CUDA - 11.7
                                          GPU - GeForce RTX 2080 Ti
Command -- 
```bash
HOROVOD_WITHOUT_GLOO=1 HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_OPERATIONS=NCCL HOROVOD_NCCL_INCLUDE=/home/anurag1/local/src/nccl-2.16.2-1/build/include HOROVOD_NCCL_LIB=/home/anurag1/local/src/nccl-2.16.2-1/build/lib  pip install --no-cache-dir horovod
```
Error --

![hvd_1](https://user-images.githubusercontent.com/10074710/210065284-aa66c696-b9c0-4c24-b84e-65fa9d1a7dbc.png)

![hvd_2](https://user-images.githubusercontent.com/10074710/210065321-e3b35e58-2bf4-427d-b00b-e7de3211a089.png)

![hvd_3](https://user-images.githubusercontent.com/10074710/210065343-21a44686-710a-4f86-ae21-eaa1bbcd6d3c.png)

**Comment by maxhgerlach** [comment]
Hi @AnuragKr,

you have configured Horovd to be built without Gloo by setting `HOROVOD_WITHOUT_GLOO=1`. But is MPI available as an alternative? You need either Gloo or MPI for Horovod.

Your error message can be understood if neither `HAVE_MPI` nor `HAVE_GLOO` are defined when building Horovod. Then no `Initialize()` member function will be available in this class:
https://github.com/horovod/horovod/blob/master/horovod/common/process_set.h#L59-L72

**Reply by AnuragKr** [reply]
Thanks, @maxhgerlach  for providing a detailed response. I again compiled HOROVOD with HOROVOD_WITH_MPI=1  and it worked.  For checking the installation I used the following command --
```bash
horovodrun --check-build
```
Output --
![Screenshot 2023-01-10 at 12 55 47 PM](https://user-images.githubusercontent.com/10074710/211488831-6c6e6402-42fd-43fc-adb9-a956e9cd60c0.png)

The Output is getting keeps on printing not stopping and I don't have GPU with TensorRT capabilities I already set TF_ENABLE_ONEDNN_OPTS=0.

**Reply by maxhgerlach** [reply]
Please post your complete install command line. Details really matter.

Also please copy-paste your error messages as text, not as screen shots, so they are searchable.

I haven't seen the new problem myself, though.

**Reply by AnuragKr** [reply]
I checked with another virtual environemnt it is working fine, Seems like some problem is in installation happened.

**Comment by AnuragKr** [comment, marked as answer]
Solution - Horovod was unable to identify the MPI library that's why the problem was coming, Then I fixed the MPI issue then I was able to install HOROVOD.
